elements/fsm-builder/fsm-builder.js

```diff
@@ -0,0 +1,1005 @@
+// eslint-disable-next-line
+const FSMBuilder = (function () {
+
+  const snapToPadding = 6; // pixels
+  const hitTargetPadding = 6; // pixels
+  const smallStateSize = 25; // pixels
+  const largeStateSize = 40; // pixels
+
+  class FSMBuilder {
+
+    constructor(name, options) {
+      this.answersName = name;
+      this.fsmTypeName = options.fsmType;
+      this.alphabetList = options.alphabet;
+
+      this.nodes = [];
+      this.links = [];
+
+      this.stateNamesToHighlight = null;
+      this.transitionsToHighlight = null;
+
+      this.currentLink = null;
+      this.selectedObject = null;
+      this.movingObject = false;
+      this.shift = false;
+      this.caretVisible = true;
+      this.nodeRadius = largeStateSize;
+
+      this.canvas = document.getElementById(this.answersName + '-fsm-canvas');
```

Comment:
The linter currently complains about this. Do I have to wait for the page to load (or something like that) before accessing the global document?